src/Dockerfile

```diff
@@ -1,20 +1,28 @@
-# Build the correct binaries
+# Build all tools except cabal-fmt and packdeps.
 FROM fossa/haskell-static-alpine:ghc-9.0.2 as builder-9.0
-LABEL org.opencontainers.image.source = "https://github.com/fossas/haskell-dev-tools"
-RUN cabal update && cabal install hlint-3.4.1 && cabal install fourmolu-0.7.0.1 
-# Replace symlinks with their actual files so we can slim down the image later
-RUN cp "$(readlink -f "$(which hlint)")" "$(readlink -f "$(which fourmolu)")" /root/.cabal/bin/
 
-# cabal-fmt doesn't build for ghc-9.0 yet, so we do a little sneaky.
-# The build from the previous GHC just works, so we use the old one.
-# hlint and fourmolu are very tightly coupled with GHC versions, but not cabal-fmt.
+RUN cabal update && cabal install --install-method=copy \
```

Comment:
Why do we need `--install-method=copy`?

Reply:
This creates a copy instead of a symlink in `.cabal/bin`. It's equivalent to your old `cp $(readlink -f)` bit.